internal/handlers/common/aggregations/stages/group.go

```diff
@@ -187,46 +187,47 @@ func (g *group) groupDocuments(ctx context.Context, in []*types.Document) ([]gro
 		}}, nil
 	}
 
-	expression, err := types.NewExpression(groupKey)
+	expression, err := aggregations.NewExpression(groupKey)
 	if err != nil {
-		var fieldPathErr *types.FieldPathError
-		if !errors.As(err, &fieldPathErr) {
+		var exprErr *aggregations.ExpressionError
+		if !errors.As(err, &exprErr) {
 			return nil, lazyerrors.Error(err)
 		}
 
-		switch fieldPathErr.Code() {
-		case types.ErrNotFieldPath:
+		switch exprErr.Code() {
+		case aggregations.ErrNotExpression:
 			// constant value aggregates values of all `in` documents into one aggregated document.
 			return []groupedDocuments{{
 				groupID:   groupKey,
 				documents: in,
 			}}, nil
-		case types.ErrEmptyFieldPath:
+		case aggregations.ErrEmptyFieldPath:
 			return nil, commonerrors.NewCommandErrorMsgWithArgument(
+				// TODO
 				commonerrors.ErrGroupInvalidFieldPath,
-				"'$' by itself is not a valid FieldPath",
+				"'$' by itself is not a valid Expression",
 				"$group (stage)",
 			)
-		case types.ErrInvalidFieldPath:
+		case aggregations.ErrInvalidExpression:
 			return nil, commonerrors.NewCommandErrorMsgWithArgument(
 				commonerrors.ErrFailedToParse,
 				fmt.Sprintf("'%s' starts with an invalid character for a user variable name", types.FormatAnyValue(groupKey)),
 				"$group (stage)",
 			)
-		case types.ErrEmptyVariable:
+		case aggregations.ErrEmptyVariable:
 			return nil, commonerrors.NewCommandErrorMsgWithArgument(
 				commonerrors.ErrFailedToParse,
 				"empty variable names are not allowed",
 				"$group (stage)",
 			)
-		case types.ErrUndefinedVariable:
+		case aggregations.ErrUndefinedVariable:
```

Comment:
Ah this is fine for `$$INVALID` which is a invalid expression, for valid expression like `$$NOW`, it should be unimplemented. Could you add this todo https://github.com/FerretDB/FerretDB/issues/2275 please.